lib/uwb/protocols/fira/UwbConfiguration.cxx

```diff
@@ -34,179 +34,226 @@ UwbConfiguration::FromDataObject(const encoding::TlvBer& /* tlv */)
 {
-    return m_sp3PhySetNumber;
+    return GetValue<uint8_t>(ParameterTag::Sp3PhySetNumber);
 }
 
 std::optional<uint8_t>
 UwbConfiguration::GetPreableCodeIndex() const noexcept
 {
-    return m_preableCodeIndex;
+    return GetValue<uint8_t>(ParameterTag::PreableCodeIndex);
 }
 
-std::optional<std::unordered_set<uwb::protocol::fira::ResultReportConfiguration>>
+std::unordered_set<ResultReportConfiguration>
 UwbConfiguration::GetResultReportConfigurations() const noexcept
 {
-    return m_resultReportConfigurations;
+    auto it = m_values.find(ParameterTag::ResultReportConfig);
+    return (it != std::cend(m_values))
+        ? std::any_cast<std::unordered_set<ResultReportConfiguration>>(it->second)
+        : std::unordered_set<ResultReportConfiguration>{};
 }
 
 std::optional<uwb::UwbMacAddressType>
 UwbConfiguration::GetMacAddressMode() const noexcept
 {
-    return m_macAddressMode;
+    return GetValue<uwb::UwbMacAddressType>(ParameterTag::MacAddressMode);
 }
 
 std::optional<uwb::UwbMacAddress>
 UwbConfiguration::GetControleeShortMacAddress() const noexcept
 {
-    return m_controleeShortMacAddress;
+    return GetValue<uwb::UwbMacAddress>(ParameterTag::ControleeShortMacAddress);
 }
 
 std::optional<uwb::UwbMacAddress>
 UwbConfiguration::GetControllerMacAddress() const noexcept
 {
-    return m_controllerMacAddress;
+    return GetValue<uwb::UwbMacAddress>(ParameterTag::ControllerMacAddress);
 }
 
 std::optional<uint8_t>
 UwbConfiguration::GetSlotsPerRangingRound() const noexcept
 {
-    return m_slotsPerRangingRound;
+    return GetValue<uint8_t>(ParameterTag::SlotsPerRr);
 }
 
 std::optional<uint8_t>
 UwbConfiguration::GetMaxContentionPhaseLength() const noexcept
 {
-    return m_maxContentionPhaseLength;
+    return GetValue<uint8_t>(ParameterTag::MaxContentionPhaseLength);
 }
 
 std::optional<uint8_t>
 UwbConfiguration::GetSlotDuration() const noexcept
 {
-    return m_slotDuration;
+    return GetValue<uint8_t>(ParameterTag::SlotDuration);
 }
 
 std::optional<uint16_t>
 UwbConfiguration::GetRangingInterval() const noexcept
 {
-    return m_rangingInterval;
+    return GetValue<uint16_t>(ParameterTag::RangingInterval);
 }
 
 std::optional<uint8_t>
 UwbConfiguration::GetKeyRotationRate() const noexcept
 {
-    return m_keyRotationRate;
+    return GetValue<uint8_t>(ParameterTag::KeyRotationRate);
 }
 
 std::optional<uwb::UwbMacAddressFcsType>
 UwbConfiguration::GetMacAddressFcsType() const noexcept
 {
-    return m_macAddressFcsType;
+    return GetValue<uwb::UwbMacAddressFcsType>(ParameterTag::MacFcsType);
 }
 
 std::optional<uint16_t>
 UwbConfiguration::GetMaxRangingRoundRetry() const noexcept
 {
-    return m_maxRangingRoundRetry;
+    return GetValue<uint16_t>(ParameterTag::MaxRrRetry);
+}
+
+const std::unordered_map<UwbConfiguration::ParameterTag, std::any>&
+UwbConfiguration::GetValueMap() const noexcept
+{
+    return m_values;
+}
+
+bool
+UwbConfiguration::operator==(const UwbConfiguration& other) const noexcept
```

Comment:
does the std::unordered_map specialization for operator== work here instead?

```suggestion
return m_values == other.m_values;
```

Reply:
Looks like the template matching can't automatically compare std::any types

Reply:
The stored type within `std::any` is erased, so the implementation can't determine which `operator==` overload to use, unfortunately. Hence this abomination of checks :'(